ui/src/assets/common.scss

```diff
@@ -164,6 +128,39 @@ button.close {
   display: flex;
   flex-direction: column;
   overflow: hidden;
+  pre,
+  code {
+    font-family: var(--monospace-font);
+  }
+  table {
+    user-select: text;
+  }
+  button.close {
+    font-family: var(--monospace-font);
+  }
+  button {
```

Comment:
I think it may not be sufficient that this declared as a descendant of the `.page` class only. For example, the top bar is not within the scope of the `.page` `<div>` element and so its error button looks like this, with a border:

![CleanShot 2023-06-12 at 14 33 11](https://github.com/eclipsesource/perfetto/assets/1615558/6c5610ec-a8d6-4114-b1be-8f359dfbbd7e)

Also, I'm not sure where else the [Button widget API](https://github.com/eclipsesource/perfetto/blob/8899422f99ba83d0f5342501b77beaef32850e62/ui/src/frontend/widgets/button.ts#L88) might be used and also be affected. In general, I think it may be necessary to double-check the reliance on the CSS classes moved in this PR in contexts outside of the `.page` `<div>`.

Maybe it would be "safer" to devise a `.perfetto` class to apply to some new top-level `<div>` element into which these definitions can be moved, that contains all of the `.topbar`, `.alerts`, and `.page` `<div>`s? This may be a dumb suggestion reflecting my relative inexperience in web programming.

Reply:
I could use descendants of "mithril-widget-container" since in our current use-case it has become Perfetto specific and its the parent of .topbar, .alerts, and .page, but I think the more correct answer would be to remove Perfetto specific data from the mithril widget and into another class that would extend it, of which I could use as a dom parent of all the Perfetto elements.

Reply:
Yeah, I think `mithril-widget-container` is something we invented for the Theia encapsulation layer and should remain in that repository. Best not to leak it into the Perfetto sources. Something defined in Perfetto that we can extend in the Theia app sounds good to me. I don't know enough about CSS and the HTML DOM to see how that would work 😕 .

Reply:
It strikes me that there must be some unique element that contains all of the Perfetto UI in a vanilla deployment of Perfetto, because the DOM is a tree. And it must be from that element that the Theia application is getting the contents to put into the `mithril-widget-container`. So perhaps the answer could be to apply the classes we're dealing with in this PR to that element, and on the Theia side _copy_ all of that element's classes to the `mithril-widget-container` so that the styling will still work for the encapsulated Perfetto?

Reply:
In Perfetto's normal environment it has the whole webpage it can edit, starting usually at the <html> tag, but the parent of .topbar, .alerts, and .page is the <main> tag.  So that's probably what we're copying, and coincidentally they don't apply any classes to the  <main> tag.

Reply:
I do agree that we need to encapsulate the styling better. We explicitly disable the main rendering in the Theia context but I don't think it should be any problem to additionally wrap the page content. We simply need to adapt the `createPage` method, i.e., replace

```
      return children;
```

with

```
      return m('.perfetto-ui', children);
```

Reply:
Changing createPage() to return a div containing the children caused the tracks to not show up, so I need to look more into this.